src/ImmediateContext.cpp

```diff
@@ -4054,15 +4054,37 @@ bool ImmediateContext::ResourceAllocationFallback(ResourceAllocationContext thre
             }
         }
     };
+
+    //If index == 0 we are checking for object deletion, else subobject deletion

```

Comment:
We could just check both, since TrimDeletedObjects will attempt to trim both, but in practice it seems like a waste of time since we are syncing on the earlier fence so it seems unlikely that the object whose fence we waited on wasn't deleted by the other subobject (or vice versa) was. Thoughts on whether or not we should change to check both anyway?

Reply:
I think what you have is fine.